### PR TITLE
A simple version of removing Markdown syntax from title

### PR DESCRIPTION
**Pull request by AlpAcA0072**
A simple version of removing Markdown syntax from title. And I'll push forward to the `markwon` method.

### CODE REVIEW

File: app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java [hunk 1]
```diff
@@ -69,9 +69,12 @@ public static String generateNoteExcerpt(@NonNull String content, @Nullable Stri
             return "";
         }
         if (!TextUtils.isEmpty(title)) {
-            final String trimmedTitle = removeMarkdown(replaceCheckboxesWithEmojis(title.trim()));
-            if (content.startsWith(trimmedTitle)) {
-                content = content.substring(trimmedTitle.length());
+            if (title != null) {
+                title = title.replaceFirst("^# ", "");
```
Comment (stefan-niedermann) [inline app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java:73]: Why didn't you replace the `# ` characters in `#removeMarkdown()`?

**Review by stefan-niedermann — CHANGES_REQUESTED** [review]

File: app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java [hunk 2]
```diff
@@ -69,9 +69,11 @@ public static String generateNoteExcerpt(@NonNull String content, @Nullable Stri
             return "";
         }
         if (!TextUtils.isEmpty(title)) {
-            final String trimmedTitle = removeMarkdown(replaceCheckboxesWithEmojis(title.trim()));
-            if (content.startsWith(trimmedTitle)) {
-                content = content.substring(trimmedTitle.length());
+            if (title != null) {
```
Comment (stefan-niedermann) [inline app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java:72]: Please remove this `if`, it is already covered by `!TextUtils.isEmpty(title)` - actually you can just revert the changes in `NoteUtil.java`.

File: markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java [hunk 3]
```diff
@@ -50,6 +50,7 @@
     private static final Pattern PATTERN_ORDERED_LIST_ITEM = Pattern.compile("^(\\d+).\\s.+$");
     private static final Pattern PATTERN_ORDERED_LIST_ITEM_EMPTY = Pattern.compile("^(\\d+).\\s$");
     private static final Pattern PATTERN_MARKDOWN_LINK = Pattern.compile("\\[(.+)?]\\(([^ ]+?)?( \"(.+)\")?\\)");
+    private static final Pattern PATTERN_MARKDOWN_TITLE = Pattern.compile("^# ");
```
Comment (stefan-niedermann) [inline markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java:53]: I am not that good in Regex, but this looks as it would just handle *exactly one* `#` character, right?
It would be a great opportunity to make it work for 1 - 6 `#` characters, so, replace:
1. `# `
2. `## `
3. `### `
4. `#### `
5. `##### `
6. `###### `

**Review by stefan-niedermann — CHANGES_REQUESTED** [review]

File: markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java (hunk 3 above)
Comment (korelstar) [reply markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java:53]: Btw. there is already a regex `PATTERN_HEADINGS` that should do the same (but better). (Why) does this not work anymore?

**Comment by AlpAcA0072** [comment]
Thanks for the advice!
I have completed it according to the above two requirements. Please check it again.

**Comment by korelstar** [comment]
Still: why adding a new regex for the same thing like an existing regex should do? The existing regex worked for a long time, and it is still working in version 3.4.1 (just tested). Are there some special cases for which it doesn't work? We should write some tests for those, then.

**Comment by stefan-niedermann** [comment]
@korelstar it has been reported [here](https://github.com/stefan-niedermann/nextcloud-notes/issues/1205) that

> When creating a note where the first line is a title like # Test, the markdown character # is included in the filename.

So something is apparently off. I didn't have time yet to have a closer look at this topic.

**Comment by korelstar** [comment]
Yeah, but I can't reproduce this. Can you?

**Comment by korelstar** [comment]
Here is already a test for this: https://github.com/stefan-niedermann/nextcloud-notes/blob/v3.4.1/markdown/src/androidTest/java/it/niedermann/android/markdown/MarkdownUtilTest.java#L660

Therefore, I think the problem is somewhere else.

**Comment by stefan-niedermann** [comment]
So @AlpAcA0072 sorry for the lost work, but as @korelstar pointed out, removing headlines already works. 

The issue affects only cases where a note is only one line long and i think it can be fixed [here](https://github.com/stefan-niedermann/nextcloud-notes/issues/1205#issuecomment-839757122) - so closing this PR. Thanks though.

**Comment by stefan-niedermann** [comment]
Fixed in https://github.com/stefan-niedermann/nextcloud-notes/commit/3bf57cd2850962045ec5cd8ff8575850770bf029

**Comment by AlpAcA0072** [comment]
Oh I see, may I ask is there any thing else I can do about Markdown?

**Comment by stefan-niedermann** [comment]
No, currently i am not aware of any Markdown related issues which are left...
